pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java

```diff
@@ -209,9 +209,9 @@ public static void buildLowLevelRealtimeIdealStateFor(String realtimeTableName,
     final String topicName = kafkaMetadata.getKafkaTopicName();
     final PinotLLCRealtimeSegmentManager segmentManager = PinotLLCRealtimeSegmentManager.getInstance();
     final int nPartitions = getPartitionsCount(kafkaMetadata);
-    // TODO Get/Infer the initial offset from table Config?
-    final long startOffset = 0L;
-    segmentManager.setupHelixEntries(topicName, realtimeTableName, nPartitions, realtimeInstances, nReplicas, 0L,
+
+    segmentManager.setupHelixEntries(topicName, realtimeTableName, nPartitions, realtimeInstances, nReplicas,
+        kafkaMetadata.getKafkaConsumerProperties().get("auto.offset.reset"), kafkaMetadata.getBootstrapHosts(),
```

Comment:
I'll do this refactoring in a different PR, so that we avoid mixing code changes with refactorings.